controllers/metal3.io/hostfirmwaresettings_controller.go

```diff
@@ -123,16 +124,12 @@ func (r *HostFirmwareSettingsReconciler) Reconcile(ctx context.Context, req ctrl
 	hfs := &metal3v1alpha1.HostFirmwareSettings{}
 	info := &rInfo{log: reqLogger, hfs: hfs, bmh: bmh}
 	if err = r.Get(ctx, req.NamespacedName, hfs); err != nil {
+		// Return error as this is unexpected
 		if k8serrors.IsNotFound(err) {
-			// A resource doesn't exist, create one
-			if err = r.newHostFirmwareSettings(info, req.NamespacedName); err != nil {
-
-				return ctrl.Result{}, errors.Wrap(err, "could not create firmware settings")
-			}
-		} else {
-			// Error reading the object - requeue the request.
-			return ctrl.Result{}, errors.Wrap(err, "could not load host firmware settings")
+			return ctrl.Result{}, errors.Wrap(err, "could not create firmware settings")
```

Comment:
Shouldn't be the error message something like `not found` (given that the creation has been moved within the BMH controller)? I'd expect to end up here if someone deletes the HFS resource before the next reconciliation loop is triggered. In such case it should be sufficient to wait enough so that the BMH controller will re-create it

Reply:
Yeah that error message needs updating, fixed.